src/main/java/com/lpvs/entity/report/LPVSReportBuilder.java

```diff
@@ -0,0 +1,482 @@
+/**
+ * Copyright (c) 2024, Samsung Electronics Co., Ltd. All rights reserved.
+ *
+ * Use of this source code is governed by a MIT license that can be
+ * found in the LICENSE file.
+ */
+package com.lpvs.entity.report;
+
+import com.lpvs.LicensePreValidationService;
+import com.lpvs.entity.LPVSFile;
+import com.lpvs.entity.LPVSLicense;
+import com.lpvs.service.LPVSLicenseService;
+import io.micrometer.common.util.StringUtils;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.thymeleaf.TemplateEngine;
+import org.thymeleaf.context.Context;
+
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * A class responsible for building reports based on the results of license scanning.
+ */
+@Component
+public class LPVSReportBuilder {
+
+    /**
+     * Creates a new instance of the LPVSReportBuilder class.
+     *
+     * @param templateEngine the template engine to use for generating reports
+     */
+    @Autowired
+    public LPVSReportBuilder(TemplateEngine templateEngine) {
+        this.templateEngine = templateEngine;
+    }
+
+    /**
+     * The template engine to use for generating reports.
+     */
+    private final TemplateEngine templateEngine;
+
+    /**
+     * The type of license detection scanner.
+     */
+    @Value("${scanner:scanoss}")
+    private String scannerType;
+
+    private final SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+    private Map<String, GroupInfo<Map<String, GroupInfo<Map<String, GroupInfo<List<LPVSFile>>>>>>>
```

Comment:
I'd suggest to create custom classes for this structure:
'ComponentInfo'
'AccessInfo'
'LicenseInfo'